packages/medusa/src/services/discount-condition.ts

```diff
@@ -0,0 +1,211 @@
+import { UpsertDiscountConditionInput } from "../types/discount"
+import { PostgresError } from "../utils/exception-formatter"
+import DiscountService from "./discount"
+
+/**
+ * Provides layer to manipulate discounts.
+ * @implements {BaseService}
+ */
+class DiscountConditionService extends BaseService {
+  private manager_: EntityManager
+  private discountRuleRepository_: typeof DiscountRuleRepository
+  private discountService_: DiscountService
+  private discountConditionRepository_: typeof DiscountConditionRepository
+  private eventBus_: EventBusService
+
+  constructor({
+    manager,
+    discountRuleRepository,
+    discountService,
+    discountConditionRepository,
+    eventBusService,
+  }) {
+    super()
+
+    /** @private @const {EntityManager} */
+    this.manager_ = manager
+
+    /** @private @const {DiscountRuleRepository} */
+    this.discountRuleRepository_ = discountRuleRepository
+
+    /** @private @const {DiscountService} */
+    this.discountService_ = discountService
+
+    /** @private @const {DiscountConditionRepository} */
+    this.discountConditionRepository_ = discountConditionRepository
+
+    /** @private @const {EventBus} */
+    this.eventBus_ = eventBusService
+  }
+
+  withTransaction(transactionManager: EntityManager): DiscountConditionService {
+    if (!transactionManager) {
+      return this
+    }
+
+    const cloned = new DiscountConditionService({
+      manager: transactionManager,
+      discountRuleRepository: this.discountRuleRepository_,
+      discountService: this.discountService_,
+      discountConditionRepository: this.discountConditionRepository_,
+      eventBusService: this.eventBus_,
+    })
+
+    cloned.transactionManager_ = transactionManager
+    cloned.manager_ = transactionManager
+
+    return cloned
+  }
+
+  resolveConditionType_(data: UpsertDiscountConditionInput):
+    | {
+        type: DiscountConditionType
+        resource_ids: string[]
+      }
+    | undefined {
+    switch (true) {
+      case !!data.products?.length:
+        return {
+          type: DiscountConditionType.PRODUCTS,
+          resource_ids: data.products!,
+        }
+      case !!data.product_collections?.length:
+        return {
+          type: DiscountConditionType.PRODUCT_COLLECTIONS,
+          resource_ids: data.product_collections!,
+        }
+      case !!data.product_types?.length:
+        return {
+          type: DiscountConditionType.PRODUCT_TYPES,
+          resource_ids: data.product_types!,
+        }
+      case !!data.product_tags?.length:
+        return {
+          type: DiscountConditionType.PRODUCT_TAGS,
+          resource_ids: data.product_tags!,
+        }
+      case !!data.customer_groups?.length:
+        return {
+          type: DiscountConditionType.CUSTOMER_GROUPS,
+          resource_ids: data.customer_groups!,
+        }
+      default:
+        return undefined
+    }
+  }
+
+  async upsertCondition(
```

Comment:
upsert meaning create or update it seams that we only create the condition am i missing something ? 😅

Reply:
If you provide `id` to the input, we [update the condition](https://github.com/medusajs/medusa/blob/fa0796a533322f704e3eb4f0e1a69400806d293a/packages/medusa/src/services/discount-condition.ts#L146) instead of creating it.